pyfr/integrators/dual/pseudo/pseudocontrollers.py

```diff
@@ -103,24 +112,28 @@ def __init__(self, *args, **kwargs):
         tplargs['maxf'] = self.cfg.getfloat(sect, 'max-fact', 1.01)
         tplargs['minf'] = self.cfg.getfloat(sect, 'min-fact', 0.98)
         tplargs['saff'] = self.cfg.getfloat(sect, 'safety-fact', 0.8)
-        tplargs['dtau_maxf'] = self.cfg.getfloat(sect, 'pseudo-dt-max-mult',
-                                                 3.0)
+        
+        dtau_minf = self.cfg.getfloat(sect, 'pseudo-dt-min-mult', 1.0)
+        dtau_maxf = self.cfg.getfloat(sect, 'pseudo-dt-max-mult', 3.0)
+
+        self._dtau_min = dtau_minf * self._dtau
+        self._dtau_max = dtau_maxf * self._dtau
+        self.alpha = 1.0
 
         if not tplargs['minf'] < 1 <= tplargs['maxf']:
             raise ValueError('Invalid pseudo max-fact, min-fact')
 
-        if tplargs['dtau_maxf'] < 1:
-            raise ValueError('Invalid pseudo-dt-max-mult')
-
-        # Limits for the local pseudo-time-step size
-        tplargs['dtau_min'] = self._dtau
-        tplargs['dtau_max'] = tplargs['dtau_maxf'] * self._dtau
+        if not dtau_minf <= 1 < dtau_maxf:
+            raise ValueError('Invalid pseudo-dt-max-mult, pseudo-dt-min-mult')
 
         # Register a kernel to compute local error
         self.backend.pointwise.register(
             'pyfr.integrators.dual.pseudo.kernels.localerrest'
         )
 
+        # Storage for kernels that require dtau-min and dtau-max at runtime
+        self.ele_scal_upts_locs = []
```

Comment:
Not sure this should be needed.

Reply:
Are the required kernels stored in another list?

I think I tried to separate out the for loop I am using it in. `self.system.ele_map` is deleted later on, so I couldn't loop over `self.system.ele_map.values()`.

Reply:
You should not need to store this at all.  Just iterate through self.pintgkernels and look for those that start with `localerrest`.